### ISSUE TITLE
Failing to call `configure` results in a segfault

### ISSUE THREAD

**Issue by olivia-fl** [reyk/rust-libtls#2]
The current safe API allows the library user to cause a segfault by trying to use `Tls::connect` without calling `Tls::configure` first. I'm not sure what the best design to avoid this is, but it caused me a lot of trouble when I didn't realize that `Tls::reset` also removes the config. Ideally the safe api exposed in the rust bindings would prevent this type of use at compile time.

**Comment by reyk** [comment]
You’re right, thanks for the report. I want to stay close to the libtls API but such unsafe behavior should be avoided with an internal state.
